BossMod/Modules/Shadowbringers/HuntA/Sugaar.cs

```diff
@@ -0,0 +1,191 @@
+        NumbingNoiseRotation = 18100, // rotation, pulls player in from 30 by max 25 units between hitboxes, 3 attacks of NumbingNoise2
+        NumbingNoiseDuringRotation = 18098, // during rotation, 0.3s cast time, 13 range, 120 degree cone
+        TailSnapRotation = 18101, // rotation, pulls player in from 30 by max 25 units between hitboxes, 3 attacks of TailSnap2
+        TailSnapDuringRotation = 18099, // Boss->self (behind), 0.3s cast, range 18, 120 degree cone
+    }
+
+    class NumbingNoise : Components.SelfTargetedAOEs
+    {
+        public NumbingNoise() : base(ActionID.MakeSpell(AID.NumbingNoise), new AOEShapeCone(13,60.Degrees())) { } 
+    }
+    class TailSnap : Components.SelfTargetedAOEs
+    {
+        public TailSnap() : base(ActionID.MakeSpell(AID.TailSnap), new AOEShapeCone(18,60.Degrees())) { }
+    }
+    class Rotation : Components.GenericAOEs
+    {
+        private int _remainingHits = 0;
+        private int _numbingnoise = 0;
+        private int _tailsnap = 0;
+        private Angle _RotationDir;
+        private Angle _RotationDirIncrement;
+
+        private static AOEShapeCone _shapeNumbingNoise = new(13, 60.Degrees());
+        private static AOEShapeCone _shapeTailSnap = new(18, 60.Degrees());
+
+        public override IEnumerable<AOEInstance> ActiveAOEs(BossModule module, int slot, Actor actor)
+        {
+            if (_remainingHits > 0 && _numbingnoise > 0)
+                yield return new(_shapeNumbingNoise, module.PrimaryActor.Position, _RotationDir);
+            if (_remainingHits > 0 && _tailsnap > 0)
+                yield return new(_shapeTailSnap, module.PrimaryActor.Position, _RotationDir);
+        }
+
+        public override void Update(BossModule module)
+        {
+            if (module.PrimaryActor.CastInfo == null || !module.PrimaryActor.CastInfo.IsSpell())
+                return;
+            switch ((AID)module.PrimaryActor.CastInfo.Action.ID)
+            {
+                case AID.NumbingNoiseRotation:
+                    _RotationDir = module.PrimaryActor.Rotation;
+                    break;
+                case AID.TailSnapRotation:
+                    _RotationDir = module.PrimaryActor.Rotation+180.Degrees();
+                    break;
+            }
+        }
+
+        public override void DrawArenaBackground(BossModule module, int pcSlot, Actor pc, MiniArena arena)
+        {
+            base.DrawArenaBackground(module, pcSlot, pc, arena);
+            if (_remainingHits > 1 && _RotationDirIncrement.Rad != MathF.PI && _numbingnoise == 1)
+                arena.ZoneCone(module.PrimaryActor.Position, 0, _shapeNumbingNoise.Radius, _RotationDir + _RotationDirIncrement, 60.Degrees(), ArenaColor.Danger);
+            if (_remainingHits > 1 && _RotationDirIncrement.Rad != MathF.PI && _tailsnap == 1)
+                arena.ZoneCone(module.PrimaryActor.Position, 0, _shapeTailSnap.Radius, _RotationDir + _RotationDirIncrement, 60.Degrees(), ArenaColor.Danger);
```

Comment:
Why do you draw this manually, rather than by reporting these aoes in ActiveAOEs?

Reply:
because you did it this way in the Burfurlur module

Reply:
Gurangatch module i mean

Reply:
changed it as requested